### DISCUSSION TITLE
Need to move installation to a new host and upgrade from 2.1.27.36 to 3.1.2 on Windows Server 2019

### DISCUSSION THREAD

**Discussion by rk0251** [nilsteampassnet/TeamPass#4171] [Q&A]
I am doing following steps:
1) created SQL dump of my live database.
2) Imported that SQL dump onto my new machine database.
3) After that I have followed the documentation and copied necessary files over to the new TeamPass folder.
4) I start the upgrade process and it is throwing me error.

PHP Fatal error:  Uncaught Defuse\\Crypto\\Exception\\BadFormatException: Encoded data is shorter than expected. in C:\\Apache24\\htdocs\\TeamPass-master\\vendor\\defuse\\php-encryption\\src\\Encoding.php:233\nStack trace:\n#0 C:\\Apache24\\htdocs\\TeamPass-master\\vendor\\defuse\\php-encryption\\src\\Key.php(53): Defuse\\Crypto\\Encoding::loadBytesFromChecksummedAsciiSafeString('\\xDE\\xF0\\x00\\x00', Object(SensitiveParameterValue))\n#1 C:\\Apache24\\htdocs\\TeamPass-master\\sources\\main.functions.php(127): Defuse\\Crypto\\Key::loadFromAsciiSafeString(Object(SensitiveParameterValue))\n#2 C:\\Apache24\\htdocs\\TeamPass-master\\sources\\main.functions.php(211): cryption('def502005cb13b1...', false, 'decrypt', Array)\n#3 C:\\Apache24\\htdocs\\TeamPass-master\\sources\\main.functions.php(4313): defuseReturnDecrypted('def502005cb13b1...', Array)\n#4 C:\\Apache24\\htdocs\\TeamPass-master\\sources\\main.functions.php(64): loadClasses('DB')\n#5 C:\\Apache24\\htdocs\\TeamPass-master\\install\\upgrade_ajax.php(202): require_once('C:\\\\Apache24\\\\htd...')\n#6 {main}\n  thrown in C:\\Apache24\\htdocs\\TeamPass-master\\vendor\\defuse\\php-encryption\\src\\Encoding.php on line 233, referer: http://localhost/TeamPass-master/install/upgrade.php

Any advice on what this means? and how to resolve the issue? Thank You.

**Comment by riharvey414** [comment]
I found this in the forums and it worked for me with some minor tweaks : 

_**For upgrade process from 2.1.27.36 i have used this method:

Start from 3.0.0.23 : https://github.com/nilsteampassnet/TeamPass/archive/refs/tags/3.0.0.23.tar.gz

Rename teampass folder old (2.1.27.36) to teampass_v2
Unzip 3.0.0.23.tar.gz
Rename unzipped folder to "teampass"
Create avatars folder in new teampass folder

copy following file:
cp -p /var/www/html/teampass_v2/includes/config/settings.php /var/www/html/teampass/includes/config/
cp -p /var/www/html/teampass_v2/includes/config/tp.config.php /var/www/html/teampass/includes/config/
cp -p /var/www/html/teampass_v2/includes/libraries/csrfp/libs/csrfp.config.php /var/www/html/teampass/includes/libraries/csrfp/libs/
cp -rp /var/www/html/teampass_v2/avatars/* /var/www/html/teampass/includes/avatars/
cp -rp /var/www/html/teampass_v2/files/* /var/www/html/teampass/files/
cp -rp /var/www/html/teampass_v2/upload/* /var/www/html/teampass/upload/
cp -rp /var/www/html/teampass_v2/Saltkey/* /var/www/html/teampass/Saltkey/ (this folder must be created before copy file)
Setting permission:
chown -R www-data.www-data /var/www/html/teampass

chmod 0777 teampass/includes
chmod -R 0777 teampass/includes/config
chmod -R 0777 teampass/includes/avatars
chmod -R 0777 teampass/includes/libraries/csrfp/libs
chmod -R 0777 teampass/includes/libraries/csrfp/log
chmod -R 0777 teampass/includes/libraries/csrfp/js
chmod -R 0777 teampass/backups
chmod -R 0777 teampass/files
chmod -R 0777 teampass/install
chmod -R 0777 teampass/upload

I have to set to 0 the column complexity_level in teampass_items table qith this query:
UPDATE teampass_items SET complexity_level = '0' WHERE complexity_level IS NULL;

Launch the upgrade from browser if obtain error open file include/config/settings.php and add
define('SECUREFILE', 'teampass-seckey.txt');
define("DB_ENCODING", "utf8");

At the end of upgrade process:
Take note of all keys generated click oon button show/hide keys and send to users

Comment block code in index.php
/*
// Quick major version check -> upgrade needed?
if (isset($SETTINGS['cpassman_version']) === true && version_compare(TP_VERSION, $SETTINGS['cpassman_version']) > 0) {
// Perform redirection
if (headers_sent()) {
echo '<script language="javascript" type="text/javascript">document.location.replace("install/install.php");</script>';
} else {
header('Location: install/upgrade.php');
}
// No other way, we should stop processing further
exit;
}
*/

restart apache and php-fpm

Enter with admin user and do 2 actions:
Rebuild configuration file
Reload cache table

Configure LDAP and sync users (if needed)

If at login of the user teampass says to put old password and current you must insert the same ldap passowrd if the local users was ldap user before upgrade**_


Thats obviously in linux, so just tweak it for windows.

**Comment by riharvey414** [comment]
Launch the upgrade from browser if obtain error open file include/config/settings.php and add
define('SECUREFILE', 'teampass-seckey.txt');
define("DB_ENCODING", "utf8");

make sure you do this step.
